### PR TITLE
Curvefi fixes (trying pr 2550 again) - draft

### PR DESCRIPTION
**Pull request by andrewhong5297**
Brief comments on the purpose of your changes:

**For Dune Engine V2**

I've checked that:

### General checks:
* [ ] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [ ] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [ ] if adding a new model, I added a test
* [ ] the filename is unique and ends with .sql
* [ ] each sql file is a select statement and has only one view, table or function defined  
* [ ] column names are `lowercase_snake_cased`
* [ ] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [ ] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [ ] `coin_id` represents the ID of the coin on coinpaprika.com
* [ ] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [ ] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [ ] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [ ] where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [ ] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

**Comment by andrewhong5297** [comment]
the ci isn't running in the right order, but I've compiled view_pools and ethereum_traces into this query and checked the cases from earlier such that there are no trades with wildly high USD amount values https://dune.com/queries/1935772

**Comment by andrewhong5297** [comment]
summary of issue is also added to readme, when you do an underlying exchange for a base token (coin0) from one of the undercoins, the event emitted uses the y pool token (which is 18 decimals) but maps to the undercoin address. So we solve this by adding a swap_type column to flag this before we divide decimals.

**Comment by andrewhong5297** [comment]
@jeff-dude just pinging for when you get the chance!

**Comment by jeff-dude** [comment]
> @jeff-dude just pinging for when you get the chance!

the CI failing is an issue -- i see you mention above, but if it fails here, it'll fail in prod. further investigation will be need done as to why that column is missing

**Comment by andrewhong5297** [comment]
> > @jeff-dude just pinging for when you get the chance!
> 
> the CI failing is an issue -- i see you mention above, but if it fails here, it'll fail in prod. further investigation will be need done as to why that column is missing

it's failing because the CI is running the diffs in the wrong order. view_pools model needs to be run first (since it has the changes to coins/undercoins columns).

File: models/curvefi/ethereum/curvefi_ethereum_view_pools.sql [hunk 1]
```diff
@@ -292,6 +260,8 @@ SELECT
     undercoin1,
     undercoin2,
     undercoin3,
+    array(coin0, coin1, coin2, coin3) as coins,
+    array(undercoin0, undercoin1, undercoin2, undercoin3) as undercoins,
```
Comment (andrewhong5297) [inline models/curvefi/ethereum/curvefi_ethereum_view_pools.sql:264]: @jeff-dude here's where the missing columns are added, but the CI isn't running this model first so the old version of this model referenced in trades is missing the column.

File: models/curvefi/ethereum/curvefi_ethereum_schema.yml [hunk 2]
```diff
@@ -102,13 +102,17 @@ models:
       - *undercoin1
       - *undercoin2
       - *undercoin3
+      - name: coins 
+        description: "All of the coins aggregated into an array, useful for exchange trades"
+      - name: undercoins 
+        description: "All of the undercoins aggregated into an array, useful for exchange_underlying trades"
```
Comment (andrewhong5297) [inline models/curvefi/ethereum/curvefi_ethereum_schema.yml:108]: defined in schema too

**Comment by andrewhong5297** [comment]
@jeff-dude okay so I reordered one line and committed to make view_pools run first, looks like ci has some diff order dependency. 

Checked cases and also for trades greater than $10m, only 6 of them. No trades greater than $100m. https://dune.com/queries/1925270

File: models/curvefi/ethereum/curvefi_ethereum_view_pools.sql [hunk 3]
```diff
@@ -292,6 +260,8 @@ SELECT
     undercoin1,
     undercoin2,
     undercoin3,
+    array(undercoin0, undercoin1, undercoin2, undercoin3) as undercoins,
+    array(coin0, coin1, coin2, coin3) as coins, --changing order to hopefully reset the CI
```
Comment (jeff-dude) [inline models/curvefi/ethereum/curvefi_ethereum_view_pools.sql:264]: do you want to move this back now? how much does order matter?
in general, editing a model is a good way to get it to trigger via CI, so good call there. just want to make sure you don't need the order back. you can also just add a comment or spacing to get it triggered

**Review by jeff-dude — APPROVED** [review]

**Pull request merged**